### DISCUSSION TITLE
How to add an extension to Animation.Channel.Target (KHR_animation_pointer)

### DISCUSSION THREAD

**Discussion by marwie** [donmccurdy/glTF-Transform#613] [Q&A] [answered]
Hello,

I'm looking at how to pass through [KHR_animation data](https://github.com/ux3d/glTF/tree/extensions/KHR_animation_pointer/extensions/2.0/Khronos/KHR_animation_pointer#examples) written to ``animations[i].channels[n].target.extensions``.

I'm not sure how to correctly set an extension on a channel target. It looks like there is currently no support for that? ([docs](https://gltf-transform.donmccurdy.com/classes/core.animationchannel.html)) 

Something like:

```js
function read(context){
    
    context.jsonDoc.json.animations?.forEach((animDef, index) => {
        animDef.channels.forEach((channel, channelIndex) => {
            const animation = context.animations[index];
            const animationChannel = animation.channels[channelIndex];
            
            const target = channel.target;
            const extensionDef = target.extensions && target.extensions[EXTENSION_NAME];
            const extension = this.document.createExtension(OpaqueExtension);
            const ext = extension.create(extensionDef, context);
            animationChannel.target.setExtension(EXTENSION_NAME, ext); // < setting the extension on the target
        });
    });
}
```

Thanks!

**Comment by donmccurdy** [comment, marked as answer]
The [AnimationChannel](https://gltf-transform.donmccurdy.com/classes/core.animationchannel.html) class inlines "target.path" and "target.node" into the class as "targetPath" and "targetNode", so I think it'd make sense to attach this extension to the channel. It can still write to the target object when serialized to JSON.
